src/spec.c

```diff
@@ -2345,6 +2345,7 @@ int IndexSpec_CreateFromRdb(RedisModuleCtx *ctx, RedisModuleIO *rdb, int encver,
     // setting unique id to zero will make sure index will not be removed from global
     // cursor map and aliases.
     sp->uniqueId = 0;
+    SchemaPrefixes_RemoveSpec(spec_ref);
```

Comment:
```suggestion
    // Remove the new spec from the global prefixes dictionary.
    // This is the only global structure that we added the new spec to at this point
    SchemaPrefixes_RemoveSpec(spec_ref);
```

Reply:
What about aliases? (`AliasTable_g`) https://github.com/RediSearch/RediSearch/blob/2b6d08d618692587bb2afe884a057fb958ca89d9/src/spec.c#LL2330C19-L2330C19

Reply:
If we succeeded to add an alias to the new spec (it was different than any previous alias), it should also be freed. It might be another bug if this scenario is possible, but even before with `sp->uniqueId = 0;` it wouldn't be freed.

Reply:
@GuyAv46 
So not calling `IndexSpec_ClearAliases` here - OK so `AliasTable_g` is not deleting the alias from its dict - OK since the original/old index has the same aliases (which should remain in the dict)
But then where is `spec->aliases` being freed from the new/redundant index? It is only cleaned from `AliasTable_Del`.
There should be a function that frees the index `aliases` member.

Let's handles cleanup of aliases in another PR?
You also noticed a potential leak in `IndexSpec_ClearAliases` with the local `QueryError`.

Reply:
@GuyAv46 Assuming all aliases that were added to global `AliasTable_g`, when being loaded again from same rdb will all fail due to
"Alias already exists", and global `AliasTable_g` will not be modified in this scenario.
I will commit you comment.